qualtran/bloqs/generalized_qsp.py

```diff
@@ -239,9 +260,11 @@ def safe_angle(x):
 class GeneralizedQSP(GateWithRegisters):
```

Comment:
Let's create a new subpackage for `qualtran/bloqs/qsp/` and move the generalized qsp code inside the new folder. We can have two files right now

1. `qualtran/bloqs/qsp/generalized_qsp.py` - This should have `GeneralizedQSP`, `gqsp_phase_factors`, `gqsp_complementary_polynomial` (rename `qsp` to `gqsp` in the last 2 functions)
2. `qualtran/bloqs/qsp/polynomial_approximations.py` - This file can have functions which return coefficeints for commonly used polynomial approximations. For example: `approx_exp_cos(precision,  scaling_const)`  and `approx_exp_sin(precision,  scaling_const)`, which returns coefficients of the Jacobi-anger polynomial approximation which can be called directly from the HamiltonianSimulationViaGQSP Bloq. 


For `HamiltonianSimulationByGQSP`, we can create a new subpackage for `bloqs/hamiltonian_simulation/` and move `HamiltonianSimulationByGQSP` there. But this can be done once the above refactorings are complete to avoid merge conflicts.

Reply:
For `GeneralizedQSP`, add a `bloq_example`, autogenerate the notebook from docstrings (you can add manually add more cells and they'll be preserve during the autogeneration process), add `bloq_autotester` tests and `assert_decompose_is_consistent_with_t_complexity`. 

Does it make sense to show some kind of symbolic analysis in GQSP?  The current API won't support it since we expect `P` and `Q` to be tuples of complex numbers and we currently don't have a way to specify sequences of variable symbolic lengths but we can file an issue to track this if there's a nice example where users would benefit from seeing the cost of applying GQSP as in terms of symbolic parameters they care about (for example hamiltonian simulation).